### PR TITLE
do not normatively specify application delivery order

### PR DESCRIPTION
**Pull request by mcmanus**
for #252

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -1370,8 +1369,8 @@ abstraction.  Streams can be created either by the client or the server, can
 concurrently send data interleaved with other streams, and can be cancelled.
 QUIC's stream lifetime is modeled closely after HTTP/2's {{?RFC7540}}.  Streams
 are independent of each other in delivery order.  That is, data that is received
-on a stream is delivered in order within that stream, but there is no particular
-delivery order across streams.  Transmit ordering among streams is left to the
+on a stream is ordered within that stream via the frame offset, but there is no particular
+order across streams.  Transmit ordering among streams is left to the
 implementation.  QUIC streams are considered lightweight in that the creation
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: Not your change, but it seems a bit verbose to talk about how lightweight streams are 3 times in one paragraph.  Maybe we can remove the last sentence("Implementations are therefore advised...")?

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -1612,8 +1611,8 @@ sender or during delivery to the application at the receiver.
 When new data is to be sent on a stream, a sender MUST set the encapsulating
 STREAM frame's offset field to the stream offset of the first byte of this new
 data.  The first byte of data that is sent on a stream has the stream offset 0.
-A receiver MUST ensure that received stream data is delivered to the application
-as an ordered byte-stream.  Data received out of order MUST be buffered for
+A receiver MUST ensure that received stream data creates an ordered
+byte-stream.  Data received out of order may need to be buffered for
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: I believe the purpose of the MUST was to ensure that if a packet with a stream frame is acked, it's not discarded.  The new wording is somewhat ambiguous to me, though I know what you're aiming for.

Also, can we remove "as long as it is not in violation of the receiver's flow control limits."  That causes a connection close error, and mentioning flow control so briefly isn't that helpful.

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (ianswett) [inline draft-ietf-quic-transport.md]: Also, we don't mention flow control anywhere in this intro section.  Possibly it should be mentioned?

Comment (martinthomson) [inline draft-ietf-quic-transport.md]: I've opened #258 to address your comments.

**Comment by MikeBishop** [comment]
OBE: #1070 addresses this.

**Pull request closed without merging**